core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceSettingService.scala

```diff
@@ -135,18 +138,27 @@ class WorkspaceSettingService(protected val ctx: RawlsRequestContext,
           }
 
           for {
-            _ <- gcsDAO.setBucketLifecycle(workspace.bucketName, googleRules)
+            _ <- gcsDAO.setBucketLifecycle(workspace.bucketName, googleRules, workspace.googleProjectId)
             _ <- workspaceSettingRepository.markWorkspaceSettingApplied(workspace.workspaceIdAsUUID,
                                                                         setting.settingType
             )
           } yield None
+
         case GcpBucketSoftDeleteSetting(GcpBucketSoftDeleteConfig(retentionDuration)) =>
           val policyBuilder = SoftDeletePolicy.newBuilder()
           policyBuilder.setRetentionDuration(Duration.ofSeconds(retentionDuration))
           val softDeletePolicy = policyBuilder.build()
 
           for {
-            _ <- gcsDAO.setSoftDeletePolicy(workspace.bucketName, softDeletePolicy)
+            _ <- gcsDAO.setSoftDeletePolicy(workspace.bucketName, softDeletePolicy, workspace.googleProjectId)
+            _ <- workspaceSettingRepository.markWorkspaceSettingApplied(workspace.workspaceIdAsUUID,
+                                                                        setting.settingType
+            )
+          } yield None
+
+        case GcpBucketRequesterPaysSetting(GcpBucketRequesterPaysConfig(enabled)) =>
+          for {
+            _ <- gcsDAO.setRequesterPays(workspace.bucketName, enabled, workspace.googleProjectId)
             _ <- workspaceSettingRepository.markWorkspaceSettingApplied(workspace.workspaceIdAsUUID,
                                                                         setting.settingType
```

Comment:
On the note of code duplication though, some of this is getting more clunky now that there are three settings. Again, feedback welcome if there are spots which could use some abstraction.